### PR TITLE
Add Rich Text Field for Create Posting Basic Info

### LINKED ISSUE #191: Add rich text input to create posting basic info form
### Context

The create posting basic form (#188) currently stores the role description in plaintext, we would like to use rich text instead.

### Acceptance Criteria
* Build a custom `RichTextField` component using the [Draft.js](https://draftjs.org/) library, supporting all the rich text features detailed in the designs
* Replace the role description `Textarea` component in `CreatePostingBasicInfo` with the new `RichTextField` component

### Technial Notes
Please refer to this [PR](https://github.com/uwblueprint/pregnancy-centre/pull/122) as an example
https://draftjs.org/

### PR DESCRIPTION
**Pull request by albertlai431**
## Ticket link

Closes #191 

## Implementation description
* Created RichTextField component with support for different font sizes, inline formatting (bold, italics, and underline), and block formatting (ordered and unordered lists)
* Replaced Textarea component with RichTextField component in the CreatePostingBasicInfo component

image

Error State:
image

## Steps to test
1. `docker-compose up --build`
2. http://localhost:3000/admin/posting/create/basic-info
3. Test out buttons for the different formatting styles and font sizes
4. Test out keyboard shortcuts for the different formatting styles (i.e., bold is Ctrl-B on Windows / Cmd-B on macOS)
5. Test out error states

## What should reviewers focus on?
* Ensuring the rich text formatting style and font size controls are functioning as expected
* Random edge cases that might break the editor

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/common/RichTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+import Italic from "../../assets/RichTextField/I.svg";
+import Underline from "../../assets/RichTextField/U.svg";
+import UnorderedList from "../../assets/RichTextField/unordered-list.svg";
+import OrderedList from "../../assets/RichTextField/ordered-list.svg";
+
+interface RichTextFieldProps {
+  initialContent: string; // content in the stringified form of a ContentState (i.e. derived from using RichTextField)
+  // this should not be reactive! (not updated onChange)
+  // how to use: for editing pre-existing content (e.g. on editing a requestGroup's description),
+  //             pass the previous description into initialContent
```
Comment (LenaNguyen) [inline frontend/src/components/common/RichTextField.tsx:33]: Move into a block comment like this

```suggestion
  /*
  content in the stringified form of a ContentState (i.e. derived from using RichTextField)
  this should not be reactive! (not updated onChange)
  how to use: for editing pre-existing content (e.g. on editing a requestGroup's description),
  pass the previous description into initialContent
  */
  initialContent: string; 
```

**Review by LenaNguyen — CHANGES_REQUESTED** [review]
This is really cool! Great work on it!

Can we easily handle/fix these scenarios?

1. Start writing in a font other than body. For example H1. Click enter to make a new line. Type "- " to start an unordered list. The selector says the font is Heading when it's actually Body. Same with ordered list.

2. Show an active state for the bold, italic, and underline buttons when the selected text has those stylings. Maybe using [this](https://stackoverflow.com/questions/48906838/is-there-any-way-to-get-the-style-under-cusor-in-draft-js) ?

1 seems like a bug but 2 is just an extra thing that does not have to be done in this ticket.

**Review by LenaNguyen — DISMISSED** [review]
Thank you for blessing the codebase with this richtext editor 🙏

**Comment by albertlai431** [comment]
@LenaNguyen had to resolve some diffs w/ `main`, could you approve again? Thanks!

**Review by LenaNguyen — APPROVED** [review]
